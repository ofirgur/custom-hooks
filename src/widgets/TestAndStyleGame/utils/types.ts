export interface GameStateContextInterface {
  gameState: GameState;
  setGameState: (gameState: Partial<GameState>) => void;
}

export interface GameState {
  gameScreen: GameScreen;
  gamePosts: GamePost[];
  gameSelectedPost?: GamePost;
}

export enum GameScreen {
  Start = 1,
  Card = 2,
  Over = 3,
}

export interface GamePost {
  userId: number,
  id: number,
  title: string,
  body: string,
}
