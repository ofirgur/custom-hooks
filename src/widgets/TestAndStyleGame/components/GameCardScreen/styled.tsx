import styled from '@emotion/styled';

export const GameCardScreenStyled = styled.div`
    color: #FFF;
    padding: 10px;
`;

export const TitleStyled = styled.h1`
    text-align: center;
    margin-bottom: 50px;
`;

export const BodyStyled = styled.section`
    text-align: center;
    margin-bottom: 50px;
`;

export const BackBotton = styled.button`
    text-align: center;
    width: 200px;
    height: 50px;
    line-height: 50px;
    background-color: #FFF;
    radius: 0;
    border-radius: 15px;
    margin: 0 auto;
    display: block;
`;