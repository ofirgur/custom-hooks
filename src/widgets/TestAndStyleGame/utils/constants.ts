import { GameState, GameScreen } from './types';

export const INITIAL_GAME_STATE: GameState = {
  gameScreen: GameScreen.game,
};