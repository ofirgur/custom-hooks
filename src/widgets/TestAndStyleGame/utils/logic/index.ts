export const _sum = (x: number, y: number) => x + y;
export const _toLowwerCase = (str: string) => str.toLocaleLowerCase(); 
export const _reduceToSum = (arr: number[]) => {
    if(!arr) return;
    return arr.reduce((acc: number, item: number) => {
        return acc += item;
    }, 0);
};

export const _sumWithInitial = (initialValue: number, arr: number[]) => arr.reduce(
    (previousValue, currentValue) => previousValue + currentValue,
    initialValue
);