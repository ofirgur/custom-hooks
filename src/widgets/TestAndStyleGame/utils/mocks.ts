
import { GameScreen } from './types';

export const mockedGameState = {
  gameScreen: GameScreen.game,
};