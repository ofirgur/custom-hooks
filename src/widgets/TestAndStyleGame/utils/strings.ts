export const STRINGS = {
  CARD_SCREEN: {
    BACK: 'BACK',
  },
};
