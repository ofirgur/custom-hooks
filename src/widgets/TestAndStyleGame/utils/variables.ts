export const VARIABLES = {
  BACKGROUND: {
    WHITE: '#FFF',
    BLACK: {
      VIEW_PORT: '#16191D',
      BUTTON: '#0E1013',
    },
    ORANGE: {
      SOLID: '#FB8948',
    },
    GRAY: '#C4C4C4',
    CREAME: '#F2D4AD',
    LAND: '#C34C23',
    DUST: '#F2D4AD',
    BEZ: '#F4D0A1',
  },
  COLOR: {
    WHITE: '#FFF',
    ORANGE: {
      SOLID: '#FB8948',
    },
    BLACK: {
      SOLID: '#000',
    },
  },
  BOX_SHADOW: {
    BUTTON: 'inset 2px 1px 10px rgba(0, 0, 0, 0.8)',
  },
  FONT_FAMILY: {
    POPPINS: 'Poppins',
  },
  FONT_STYLE: {
    NORMAL: 'font-style',
  },
  FONT_WEIGHT: {
    NORMAL: '300',
  },
  BREAK_POINTS: {
    LT_MIN: '400px',
    MD_MIN: '690px',
  },
};
