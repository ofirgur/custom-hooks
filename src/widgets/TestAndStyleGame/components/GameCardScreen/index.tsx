import React from 'react';

import { useGameStateContext } from '../../utils/context';
import { GameScreen } from '../../utils/types';
import { STRINGS } from '../../utils/strings';
import { GameCardScreenStyled, TitleStyled, BodyStyled, BackBotton } from './styled';

const GameCardScreen = () => {
  const { gameState, setGameState } = useGameStateContext();

  const handleBack = () => {
    setGameState({
      gameScreen: GameScreen.Start,
      gameSelectedPost: undefined,
    });
  };

  return (
    <GameCardScreenStyled>
      <TitleStyled>{gameState.gameSelectedPost?.title}</TitleStyled>
      <BodyStyled>{gameState.gameSelectedPost?.body}</BodyStyled>
      <BackBotton onClick={handleBack}>{STRINGS.CARD_SCREEN.BACK}</BackBotton>
    </GameCardScreenStyled>
  );
};

export default GameCardScreen;
