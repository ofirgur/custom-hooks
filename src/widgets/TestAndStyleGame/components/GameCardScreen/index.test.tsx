import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

import { mockedGameState } from '../../utils/mocks';
import GameCardScreen from './index';

const mockedSetGameState = jest.fn();
jest.mock('../../utils/context', () => {
  return {
    useGameStateContext: () => ({
      setGameState: mockedSetGameState,
      gameState: {
        ...mockedGameState,
      },
    }),
  }
});

describe('GameCardScreen', () => {
  it('should render correctly', () => {
    const { container } = render(<GameCardScreen />);
    expect(container).toMatchSnapshot();
  });

  it('should render title of the selected post', () => {
    const { getByText } = render(<GameCardScreen />);
    expect(getByText(mockedGameState.gameSelectedPost.title)).toBeVisible();
  });

  // it('should render body of the selected post', () => {
  //   const { getByText } = render(<GameCardScreen />);
  //   expect(screen.getByText(mockedGameState.gameSelectedPost.body)).toBeVisible();
  // });

  it('in case user clicks on the post card it should call to setGameState', async () => {
    const { getByRole } = render(<GameCardScreen />);
    const user = userEvent.setup();
    await user.click(getByRole('button'));
    expect(mockedSetGameState).toHaveBeenCalled();
  });
});
