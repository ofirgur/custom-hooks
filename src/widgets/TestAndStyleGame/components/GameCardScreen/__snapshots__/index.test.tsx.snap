// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCardScreen should render correctly 1`] = `
<div>
  <div
    class="css-195sjo6"
  >
    <h1
      class="css-kvywuf"
    >
      ea molestias quasi exercitationem repellat qui ipsa sit aut
    </h1>
    <section
      class="css-kvywuf"
    >
      et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut
    </section>
    <button
      class="css-ga8kd7"
    >
      BACK
    </button>
  </div>
</div>
`;
